tt_metal/CMakeLists.txt

```diff
@@ -33,8 +35,7 @@ target_precompile_headers(tt_metal PRIVATE
     <vector>
 )
 
-target_link_libraries(tt_metal PUBLIC compiler_flags linker_flags metal_header_directories $<$<BOOL:${ENABLE_TRACY}>:TracyClient>)  # linker_flags = -rdynamic if tracy enabled
-target_link_directories(tt_metal PUBLIC ${PROJECT_BINARY_DIR}/lib)    # required so tt_metal can find device library
```

Comment:
How does it find it now? Is it because it directly brings it into the CMake build via https://github.com/tenstorrent/tt-metal/pull/11496/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR118?

Reply:
That line was originally needed because we built umd with a `make flow` so cmake didn't have a target to link to. Now that umd has cmake infra we don't need to give it a directory to look in. This line could've been removed months ago oops.